From fb4ad6d6019b9d3ba95715a5f5f367f26cbf0ec5 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 08/10] health@1.0-impl: Add missing charger links to libminui
 libbase libpng and libz

The charger links against them so that additional libraries defined with
LOCAL_HAL_STATIC_LIBRARIES can consume them. However the health
implementation doesn't do so and thus fails building with the modules
---
 health/1.0/default/Android.mk | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/health/1.0/default/Android.mk b/health/1.0/default/Android.mk
index 199ab412..781e847d 100644
--- a/health/1.0/default/Android.mk
+++ b/health/1.0/default/Android.mk
@@ -20,6 +20,16 @@ LOCAL_SHARED_LIBRARIES := \
 
 LOCAL_STATIC_LIBRARIES := android.hardware.health@1.0-convert
 
+ifneq ($(strip $(LOCAL_CHARGER_NO_UI)),true)
+LOCAL_STATIC_LIBRARIES += \
+    libbase \
+    libpng \
+    libz
+
+LOCAL_WHOLE_STATIC_LIBRARIES := \
+    libminui
+endif
+
 LOCAL_HAL_STATIC_LIBRARIES := libhealthd
 
 include $(BUILD_SHARED_LIBRARY)
-- 
2.11.0

